### PR TITLE
[ssh2] Thorough update of ssh2 typings

### PR DESCRIPTION
**Pull request by dhensby**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/mscdex/ssh2/blob/master/SFTP.md
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

---

After my previous update in #60982 I've noticed there are a lot of inconsistencies between the typings and the current implementation. As such I've gone though and thoroughly updated the typings for ssh2

### CODE REVIEW

**Review by bragle — CHANGES_REQUESTED** [review]
Curious about the comments. Looks good otherwise. Feel free to add yourself as a definition owner 😸

File: types/ssh2/index.d.ts [hunk 1]
```diff
@@ -1519,334 +1319,348 @@ export interface SFTPWrapper extends events.EventEmitter {
...
+export type GetStreamCallback = (err: Error | undefined, stream: Duplex) => void;
+
 /**
  * Interface representing an inbound agent request. This is defined as an
  * "opaque type" in the ssh2 documentation, and should only be used
  * for correlation, not introspected.
  */
 export interface AgentInboundRequest {
     __opaque_type: never
+    // hasResponded(): boolean;
```
[the remark this answers is not in the document]
Comment (dhensby) [reply types/ssh2/index.d.ts:1654]: Good question - I had started defining them, but realised as the type is not meant to be used that I wouldn't bother. I'll delete them

[the remark this answers is not in the document]
Comment (dhensby) [reply types/ssh2/index.d.ts]: Done

**Review by bragle — APPROVED** [review]
Sweet!

**Comment by dhensby** [comment]
Ready to merge

**Pull request merged**